twenty-first/src/shared_math/polynomial.rs

```diff
@@ -2420,10 +2450,44 @@ mod test_polynomials {
         assert_eq!(expected_sixth_rem, actual_sixth_rem);
     }
 
+    #[test]
+    pub fn xgcd_b_field_pol_test() {
+        for _ in 0..100 {
+            let x: Polynomial<BFieldElement> = gen_polynomial_non_zero();
+            let y: Polynomial<BFieldElement> = gen_polynomial_non_zero();
+            let (gcd, a, b): (
+                Polynomial<BFieldElement>,
+                Polynomial<BFieldElement>,
+                Polynomial<BFieldElement>,
+            ) = Polynomial::xgcd(x.clone(), y.clone());
+            assert!(gcd.is_one());
```

Comment:
I think the probability of this test failing even though the gcd was computed correctly, i.e., two random polynomials of degree at most 40 sharing a factor, is smaller than $\frac{40\cdot40}{p} \approx 8.7\cdot 10^{-17}$, which is probably acceptable. Still, something in me itches…

Reply:
I'm not sure my polynomial understanding suffices to improve this test. Feel free to change it yourself though.